### ISSUE TITLE
Passing setters with use_setters_to_change_properties

### ISSUE THREAD

**Issue by maxlapides** [dart-lang/linter#1200]
This example does NOT create a lint warning:

```
class MyClass {
  bool isOn;

  // this is good!
  void _setIsOn(bool newIsOn) => isOn = newIsOn;

  AnotherClass create() {
    return AnotherClass(setIsOn: _setIsOn);
  }
}

class AnotherClass {
  Function setIsOn;
  AnotherClass({this.setIsOn});
}
```

But when I don't use the lambda expression syntax, I do get a lint warning:

```
class MyClass {
  bool isOn;

  // this is bad!
  void _setIsOn(bool newIsOn) {
    isOn = newIsOn;
  }

  AnotherClass create() {
    return AnotherClass(setIsOn: _setIsOn);
  }
}

class AnotherClass {
  Function setIsOn;
  AnotherClass({this.setIsOn});
}
```

**Comment by srawlins** [comment]
@JPaulsen , you include the `if (body is BlockFunctionBody` check in the original implementation. Is this intentional, or an oversight, to not include `ExpressionFunctionBody`?

**Comment by maxlapides** [comment]
@srawlins thanks for the fix! also, a quick question, what would be the "correct" way of doing this kind of thing?

**Comment by srawlins** [comment]
@maxlapides Oh, I hope the lint is clear about this, but instead of

```dart
class MyClass {
  bool isOn;

  // this is bad!
  void _setIsOn(bool newIsOn) => isOn = newIsOn;
}
```

You should use Dart's native setters:

```dart
class MyClass {
  bool _isOn;

  // this is good!
  set isOn(bool value) => _isOn = value;
}
```

**Comment by maxlapides** [comment]
@srawlins cool. but what if i want to pass that setter? in my original example, i was passing `_setIsOn` to another class constructor. is there a good way of doing this?

**Comment by srawlins** [comment]
I'm not sure actually... you'd have to test out if you can tear off a setter. Good question.

**Comment by maxlapides** [comment]
haha okay, at least i feel better that i'm not missing something obvious! i haven't found a way to do this other than by doing something that would violate the `use_setters_to_change_properties` lint rule.
